services/app-api/src/resolvers/healthPlanPackage/createHealthPlanPackage.ts

```diff
@@ -71,24 +79,63 @@ export function createHealthPlanPackageResolver(
             contractType: input.contractType,
         }
 
-        const pkgResult = await store.insertHealthPlanPackage(insertArgs)
-        if (isStoreError(pkgResult)) {
-            const errMessage = `Error creating a package of type ${pkgResult.code}. Message: ${pkgResult.message}`
-            logError('createHealthPlanPackage', errMessage)
-            setErrorAttributesOnActiveSpan(errMessage, span)
-            throw new GraphQLError(errMessage, {
-                extensions: {
-                    code: 'INTERNAL_SERVER_ERROR',
-                    cause: 'DB_ERROR',
-                },
-            })
-        }
+        //Here is where we flag the insert
+        if (ratesDatabaseRefactor) {
+            const contractResult = await store.insertDraftContract(insertArgs)
+            if (contractResult instanceof Error) {
+                const errMessage = `Error creating a draft contract. Message: ${contractResult.message}`
+                logError('createHealthPlanPackage', errMessage)
+                setErrorAttributesOnActiveSpan(errMessage, span)
+                throw new GraphQLError(errMessage, {
+                    extensions: {
+                        code: 'INTERNAL_SERVER_ERROR',
+                        cause: 'DB_ERROR',
+                    },
+                })
+            }
+
+            // Now we do the conversions
+            const pkg =
+                convertContractToUnlockedHealthPlanPackage(contractResult)
+
+            if (pkg instanceof Error) {
+                const errMessage = `Error converting draft contract. Message: ${pkg.message}`
+                logError('createHealthPlanPackage', errMessage)
+                setErrorAttributesOnActiveSpan(errMessage, span)
+                throw new GraphQLError(errMessage, {
+                    extensions: {
+                        code: 'INTERNAL_SERVER_ERROR',
+                        cause: 'PROTO_DECODE_ERROR',
```

Comment:
Isn't this a `ZOD_PARSE_ERROR`

Reply:
At this point, the zod parse error would happen in `toDomain` so it would be when decoding. Maybe the error checking above this one would need more work

```typescript
const contractResult = await store.insertDraftContract(insertArgs)
if (contractResult instanceof Error) {
    const errMessage = `Error creating a draft contract. Message: ${contractResult.message}`
    logError('createHealthPlanPackage', errMessage)
    setErrorAttributesOnActiveSpan(errMessage, span)
    throw new GraphQLError(errMessage, {
        extensions: {
            code: 'INTERNAL_SERVER_ERROR',
            cause: 'DB_ERROR',
        },
    })
}
```

That store function could return a prisma error or a parsing error.